### DISCUSSION TITLE
read data: chunk

### DISCUSSION THREAD

**Discussion by jakedowns** [directus/directus#16287] [Feature Requests]
in Laravel Eloquent ORM, there's the option to chunk db reads for large data sets, and then provide a callback to execute logic on each chunk

i think Read Data could benefit from the ability to query ALL items in a collection, but then process them in chunks
this would abstract away the need for messy custom paging logic
in the current set up, i guess it would require passing each chunk array off to a subsequent flow
but, it'd still be a start.

This would be great in combination with https://github.com/directus/directus/discussions/16258

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
